src/query/sql/src/planner/binder/table.rs

```diff
@@ -466,25 +469,36 @@ impl Binder {
     async fn bind_subquery(
         &mut self,
         bind_context: &mut BindContext,
+        lateral: bool,
         subquery: &Query,
         alias: &Option<TableAlias>,
     ) -> Result<(SExpr, BindContext)> {
-        // For subquery, we need to use a new context to bind it.
-        let mut new_bind_context = BindContext::with_parent(Box::new(bind_context.clone()));
-        let (s_expr, mut res_bind_context) =
-            self.bind_query(&mut new_bind_context, subquery).await?;
+        // If the subquery is a lateral subquery, we need to let it see the columns
+        // from the previous queries.
+        let (result, mut result_bind_context) = if lateral {
```

Comment:
`if !lateral`?

Reply:
If not lateral, we won't let it see the previous columns just as the current implementation.